iso15118/secc/states/iso15118_2_states.py

```diff
@@ -765,6 +768,49 @@ class PaymentDetails(StateSECC):
     def __init__(self, comm_session: SECCCommunicationSession):
         super().__init__(comm_session, Timeouts.V2G_SECC_SEQUENCE_TIMEOUT)
 
+    def _all_certificates_from_chain(
+        self, certificate_chain: CertificateChain
+    ) -> List[Certificate]:
+        """Return all certificates from a certificate chain as a list.
+
+        The order should be: leaf certificate, sub-CA 2, sub-CA 1,
+        if all three are present.
+        """
+        return [
+            certificate_chain.certificate
+        ] + certificate_chain.sub_certificates.certificates
+
+    def _get_contract_certificate_hash_data(
+        self,
+        certificate_chain: Optional[CertificateChain],
+    ) -> Optional[List[Dict[str, str]]]:
+        """Return a list of hash data for a contract certificate chain."""
+        if certificate_chain is not None:
+            return None
+
+        return [
+            derive_certificate_hash_data(certificate)
+            for certificate in self._all_certificates_from_chain(certificate_chain)
+        ]
+
+    def _build_pem_certificate_chain(
+        self, certificate_chain: Optional[CertificateChain]
+    ) -> Optional[str]:
+        """Return a string of certificates in PEM form concatenated together."""
+        if certificate_chain is None:
+            return None
+
+        return "".join(
+            [
+                certificate_to_pem_string(certificate)
+                for certificate in self._all_certificates_from_chain(certificate_chain)
+            ]
+        )
+
+    def _mobility_operator_root_cert_path(self) -> str:
+        """Return the path to the MO root.  Included to be patched in tests."""
+        return CertPath.MO_ROOT_DER
```

Comment:
do we really need this?
cant we do the patching like I did here in pyslac?
https://github.com/SwitchEV/pyslac/blob/master/tests/test_slac_session.py#L95

Reply:
Quite possibly -- I added this because it was the quickest way I saw to get the tests passing.

However, might it be better to leave this in, since it doesn't complicate things now too much, and will be useful in case we have certificates of multiple mobility operators locally (in which case we'd need to access the correct one, rather than just loading from a hardcoded path)?

Reply:
hmm, I understand your point of view, but with patching we can achieve the same thing...you can decide what to return in different tests. I just dont feel that is right to leave methods on the way of "real code" specially given that there is a solution to avoid it and keep functional code and tests in separate.

Reply:
There's more to this than patching -- at some point we may not have only one mobility root certificate, right?  Different mobility operators may ultimately derive from different roots, and we might need to have them both on the same charging station (or is this wrong?)

I probably wouldn't have added this just for that purpose (yet), but now that it's in it seems to me fine to leave it there.

If you insist, I can move to patching for tests and get rid of this.